runtime/common/ExecutionContext.h

```diff
@@ -100,6 +100,11 @@ class ExecutionContext {
   /// register after execution. Empty means no reordering.
   std::vector<std::size_t> reorderIdx;
 
+  /// @brief A buffer containing the return value of a kernel invocation.
+  // Note: this is only needed for invocation not able to return a
+  // sample_result.
+  std::vector<char> invocationResultBuffer;
```

Comment:
I'm wondering if we'll need to think about endianness issues here. OTOH, our assumption to date is that the endianness is the same.

Reply:
I added an endian check in https://github.com/NVIDIA/cuda-quantum/pull/1969/commits/026834c4a5e7382e0a8e7caf5750ceb19be21b96. 

Yea, our assumption about endianness is kind of loose at the moment, hence I think throwing an error is sufficient (rather than trying to handle it).